.circleci/config.yml

```diff
@@ -644,58 +651,91 @@ jobs:
           webhook: << parameters.slack-webhook >>
 
   wp-desktop-source:
-    <<: *desktop_machine_executor_defaults
+    docker:
+      - image: circleci/node:12.18.0-browsers
+    resource_class: medium
+    <<: *desktop_defaults
     environment:
       VERSION: << pipeline.git.tag >>
+      CHROMEDRIVER_SKIP_DOWNLOAD: 'true'
+      DETECT_CHROMEDRIVER_VERSION: 'false'
+      NODE_OPTIONS: --max-old-space-size=3072
     steps:
       - checkout
+      - run: *update-node
       - when:
           condition: << pipeline.git.tag >>
           steps:
             - run:
                 name: Ensure package.json Version And Tag Match
                 command: cd desktop/bin && node validate_tag.js $VERSION
+      - run: *desktop-decrypt-assets
       - run:
-          name: Install Node
+          name: Compute cache checksums
           command: |
-            set +e
-            set +x
-            export NVM_DIR="/opt/circleci/.nvm"
-            [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
-            nvm install
-            nvm use
+            # Get latest commit that modified packages directory
+            echo "$(git log -n 1 --format='%h' -- packages)" > packages-hash
+
+            # Get latest commit that modified .css and related files
+            echo "$(git log -n 1 --format='%h' -- '*.css' '*.scss' '*.sass')" > webpack-css-loader-hash
+      - restore_cache: *desktop-restore-yarn-cache
+      - restore_cache: *desktop-restore-packages-cache
+      - restore_cache: *desktop-restore-webpack-css-loader-cache
+      - run:
+          name: Install Linux deps
+          command: |
+            sudo apt update
+            sudo apt-get install -y git-restore-mtime bc
+      - run:
+          name: Restore file modified time
+          command: |
+            # Webpack cache-loader depends on file mtime, which is reset by "git clone".
+            # Use git-restore-mtime to restore original modification time of files based
+            # on date of most recent commit that modified them.
+            # Ref: http://manpages.ubuntu.com/manpages/bionic/man1/git-restore-mtime.1.html
+            #
+            # Note: this utility is highly optimized and is able to process all Calypso
+            # files in less than 10 seconds!
+            /usr/lib/git-core/git-restore-mtime --force --commit-time --skip-missing
       - run:
           name: Install Dependencies
           command: |
-            export NVM_DIR="/opt/circleci/.nvm"
-            [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+            set +e
+            source $HOME/.nvm/nvm.sh
             nvm use
             npm install -g yarn
 
+            # silence Chromedriver installation/resolution errors.
             yarn config set chromedriver_skip_download true
```

Comment:
I think these can be on `environment` now.

Reply:
Interestingly enough, using the `CHROMEDRIVER_` env vars in the `environment` key did not work with the `machine` executor instance (very bizarre), so I had to use yarn config instead. Very, very strange.

Reply:
Update: not sure what was missing last time but silencing chromedriver errors in `environment` key now works - update the config.